### PR TITLE
ivy: make rectangle assignments run in parallel

### PR DESCRIPTION
**Pull request by rsc**
Now that assignment can be multidimensional,
it can be expensive. Make it cheaper when there are many CPUs.

### CODE REVIEW

File: value/binary.go [hunk 1]
```diff
@@ -208,6 +209,12 @@ func init() {
 			whichType:   binaryArithType,
 			fn: [numType]binaryFn{
 				intType: func(c Context, u, v Value) Value {
+					if u.(Int) == 0 {
+						return v
+					}
+					if v.(Int) == 0 {
+						return u
+					}
```
Comment (robpike) [inline value/binary.go:217]: why are you doing this microoptimization?

File: value/binary.go [hunk 2]
```diff
@@ -252,6 +259,12 @@ func init() {
 			whichType:   binaryArithType,
 			fn: [numType]binaryFn{
 				intType: func(c Context, u, v Value) Value {
+					if u.(Int) == 1 || v.(Int) == 0 {
```
Comment (robpike) [inline value/binary.go:262]: ditto

**Comment by robpike** [comment]
Thanks. They deserve a comment or perhaps a clever helper function?

**Comment by rsc** [comment]
I've removed the Value allocation optimizations and will send them separately once I have more careful measurements.

**Review by robpike — APPROVED** [review]

**Pull request merged**